include/arch/riscv/arch/32/mode/machine.h

```diff
@@ -12,7 +12,7 @@
 
 static inline uint64_t riscv_read_time(void)
 {
-    uint32_t nH1, nL, nH2;
+    register word_t nH1, nL, nH2;
```

Comment:
The kernel doesn't use `register` keyword anywhere.  Also why do you wish to switch uint32_t to word_t as they're the same type in this context.

Reply:
- I've been digging a bit deeper into the `register` keyword and the history.Seems nowadays it's indeed better to avoid using it unless there is a really good reason. Remove this. 
- `word_t` seems semantically the better choice for case like this, that deal with register data. Furthermore, it's a pattern that is copy/paste safe. Technically it does not make any difference.